core/src/main/java/org/mskcc/cbio/portal/model/converter/MutationModelConverter.java

```diff
@@ -45,8 +45,8 @@ public ExtendedMutation convert(Mutation mutation) {
         if (mutationEvent.getEndPosition() != null) {
             event.setEndPosition(mutationEvent.getEndPosition());
         }
-        if (mutationEvent.getAminoAcidChange() != null) {
-            event.setProteinChange(mutationEvent.getAminoAcidChange());
+        if (mutationEvent.getProteinChange() != null) {
```

Comment:
+1 on checking it as early as possible (constructor) and not checking later. The few fields that may indeed be null can be turned into Optionals so it can be properly handled by the type system